stan/math/rev/core/var.hpp

```diff
@@ -308,10 +215,13 @@ class var {
    * then (a += b) behaves exactly the same way as (a = a + b),
    * creating an intermediate variable representing (a + b).
    *
+   * @tparam S a type that is convertible to `value_type`
    * @param b The variable to add to this variable.
    * @return The result of adding the specified variable to this variable.
    */
-  inline var& operator+=(var b);
+  template <typename S, require_convertible_t<S&, promote_args_t<T>>* = nullptr>
+  inline var_value<T, require_floating_point_t<T>>& operator+=(
```

Comment:
Can `require_floating_point_t<T>` be removed? That's already guaranteed by the class, right?

Reply:
I tried removing this before but got some really weird compiler errors

Reply:
Got it. I think cause the type is defined that way something needs to be here so this makes sense enough.

Reply:
Oh wait maybe I'm dumb and this is fine to remove

Reply:
I've been juked.

Reply:
Consider yourself juked! I'll push that up once I'm done staring at the stack allocator code. I'm trying to figure out a way to not have a seperate stack here but I don't see another option atm